R/helper.R

```diff
@@ -80,26 +174,34 @@ mn_sd_cov <- function(weights_item, weights_latent, alpha_r, alpha_f, psi_r,
 #' @param lower.tail A logical scalar; if TRUE (default), probabilities are 
 #' \eqn{P[X <= x]}; otherwise, \eqn{P[X > x]}. 
 #' @return The output will be the cumulative probability of q or \eqn{1 - q} on 
-#' the mixture normal distribution.
+#'   the mixture normal distribution.
 #' @examples
 #' \dontrun{
 #' pnormmix(1, 0, 3.1, 1.7, 3.1, lower.tail = FALSE)
 #' }
 
-pnormmix <- function(q, mean1 = 0, sd1 = 1, mean2 = 0, sd2 = 1, pmix1 = 0.5, 
+pnormmix <- function(q, mean1 = 0, sd1 = 1, mean2 = 0, sd2 = 1, pmix1 = 0.5,
                      lower.tail = TRUE) {
   stopifnot(pmix1 > 0, pmix1 < 1)
-  as.vector(c(pmix1, 1 - pmix1) %*% 
-              sapply(q, pnorm, mean = c(mean1, mean2), sd = c(sd1, sd2), 
+  as.vector(c(pmix1, 1 - pmix1) %*%
+              sapply(q, pnorm, mean = c(mean1, mean2), sd = c(sd1, sd2),
                      lower.tail = lower.tail))
 }
 
-# pnormmix(, mean1 = 1.53, sd1 = 0.89, mean2 = 1.32, sd2 = 0.88, pmix1 = 0.5, lower.tail = FALSE)
+
+pnormmix_mult <- function(q, means = 0, sds = 1, pmix = NULL, 
```

Comment:
I couldn't find `pnormmix_mult()` and `qnormmix_mult()` being called in other places. How should they be used?

Reply:
`pnormmix_mult()` and `qnormmix_mult()` are the updated functions that handle multiple groups and are used for the current version of PartInv. `pnormmix` and `qnormmix` are old versions that we might want to keep for the deprecated version of PartInv which depends on these two functions.

Reply:
Okay I see them being used now. We may also just rewrite `pnormmix()` and `qnormmix()` so that they internally call `pnormmix_mult()` and `qnormmix_mult()`.

Reply:
Sounds good. I have made note of this and will update the functions accordingly in a future PR.